src/SearchParameters/shortener.js

```diff
@@ -81,5 +88,22 @@ module.exports = {
    */
   encode: function(key) {
     return keys2Short[key];
+  },
+  /**
+   * Set an attribute / short attribute pair
+   * @param {string} key the attribute
+   * @param {string} shortKey the shorten attribute
+   */
+  set: function(key, shortKey) {
+    var oldShort = keys2Short[key];
+    var idx = ENCODED_PARAMETERS.indexOf(oldShort);
+    if (idx === -1) {
+      ENCODED_PARAMETERS.push(shortKey);
+    } else {
+      ENCODED_PARAMETERS[idx] = shortKey;
+      delete short2Keys[oldShort];
```

Comment:
Exactly @bobylito , here we replace it in `keys2Short`, so there's no reason it stays in `short2Keys` which is supposed to be `invert(keys2Short)`.